packages/scheduler-prisma/src/helpers.ts

```diff
@@ -17,39 +17,88 @@ const SIX_MONTHS_IN_MS = 15778476000;
 export const isHealthy = (heartbeat: number, timeout: number) =>
   new Date().getTime() - timeout < heartbeat;
 
+const getValidRule = (recurrence: string, timezone?: string) => {
+  const isICalRule = recurrence.includes('DTSTART');
+  if (isICalRule) return recurrence;
+
+  let derivedTimezone = timezone ?? 'Australia/Sydney';
+  const rule = recurrence
+    .split(';')
+    .filter(b => {
+      const [key, value] = b.split('=');
+      if (key === 'TZID') {
+        derivedTimezone = value;
+      }
+      return key !== 'TZID';
+    })
+    .join(';');
+
+  const timeISO8601 = moment().tz(derivedTimezone).format('YYYYMMDDTHHmmss');
+  return `DTSTART;TZID=${derivedTimezone}:${timeISO8601}\nRRULE:${rule}`;
+};
+
 // interval should be iCal String.
-export const computeNextRunAt = (
+export const computeNextRun = (
   interval: string,
-  timezone = 'UTC'
+  {
+    /**
+     * @description Timezone to compute the next run at
+     * @default UTC
+     */
+    timezone = 'UTC',
+    /**
+     * @description Start date to compute the next run at
+     * @default now()
+     */
+    startDate = moment().toISOString(),
+  } = {}
 ): string => {
   if (!interval) {
-    throw new Error('Invalid interval argument supplied to computeNextRunAt');
+    throw new Error('Need a valid interval to compute next run at');
   }
 
-  const isValidRule = interval.includes('DTSTART');
-
-  if (!isValidRule) {
-    const rule = interval
-      .split(';')
-      .filter(b => !b.includes('TZID'))
-      .join(';');
+  const rule = getValidRule(interval, timezone);
+  const rrule = RRuleSet.parse(rule);
+  const start = moment(startDate).valueOf();
+  const end = moment(start).add(SIX_MONTHS_IN_MS, 'ms').valueOf();
+  return new Date(rrule.between(start, end, true)[0]).toISOString();
+};
 
-    const timeISO8601 = moment().tz(timezone).format('YYYYMMDDTHHmmss');
-    const rrule = RRuleSet.parse(
-      `DTSTART;TZID=${timezone}:${timeISO8601}\nRRULE:${rule}\nEXDATE;TZID=${timezone}:${timeISO8601}`
-    );
-    return new Date(rrule.all(1)[0]).toISOString();
+export const computeNextRuns = (
+  interval: string,
+  {
+    /**
+     * @description Timezone to compute the next run at
+     * @default UTC
+     */
+    timezone = 'UTC',
+    /**
+     * @description Start date to compute the next run at
+     * @default now()
+     */
+    startDate = moment().toISOString(),
+    /**
+     * @description The number of runs to compute
+     * @default 1
+     * @max 10
+     */
+    count = 1,
+  } = {}
+): string[] => {
+  if (!interval) {
+    throw new Error('Need a valid interval to compute next run at');
   }
 
-  const rrule = RRuleSet.parse(interval);
+  const rule = getValidRule(interval, timezone);
+  const rrule = RRuleSet.parse(rule);
+  const runCount = Math.min(count, 30);
```

Comment:
Why 30? 
Also this does not match the comment.

Reply:
ah it's supposed to be 30, there's a call in OM with at least 28 runs. the comment is incorrect